evil-collection.el

```diff
@@ -175,6 +193,55 @@ Evil Collection for that mode. More arguments may be added in the future, so
 functions added to this hook should include a \"&rest _rest\" for forward
 compatibility.")
 
+(defun evil-collection-define-key (state package map-sym &rest bindings)
+  "Wrapper for `evil-define-key*' with additional features.
+Filter keys on the basis of `evil-collection-key-whitelist' and
+`evil-collection-key-blacklist'. Store bindings in
+`evil-collection-bindings-record'."
+  (declare (indent defun))
+  (let* ((whitelist (mapcar 'kbd evil-collection-key-whitelist))
+         (blacklist (mapcar 'kbd evil-collection-key-blacklist))
+         (record (cdr-safe (assoc package evil-collection-bindings-record))))
+    (while bindings
+      (let ((key (pop bindings))
+            (def (pop bindings)))
+        (when (or (and whitelist (member key whitelist))
+                  (not (member key blacklist)))
+          (push (cons key def) record)
+          (eval-after-load package
```

Comment:
Why not using `with-eval-after-load`?

Reply:
I can. I was in the habit of using the older function internally.

Reply:
I actually find it a little more straightforward to use `eval-after-load` here, because `with-eval-after-load` is a macro and doesn't eval its arguments.